lib/browser/mqtt.ts

```diff
@@ -367,14 +367,16 @@ export class MqttClientConnection extends BufferedEventEmitter {
      */
     async subscribe(topic: string, qos: QoS, on_message?: (topic: string, payload: ArrayBuffer) => void): Promise<MqttSubscribeRequest> {
         this.subscriptions.insert(topic, on_message);
-        return this.connection.subscribe(topic, { qos: qos })
-            .catch((reason: any) => {
-                this.emit('error', new CrtError(reason));
-            })
-            .then((value) => {
-                const sub = (value as ISubscriptionGrant[])[0];
-                return { topic: sub.topic, qos: sub.qos };
+        return new Promise((resolve, reject) => {
+            this.connection.subscribe(topic, { qos: qos }, (error, packet) => {
+                if (error) {
+                    reject(new CrtError(error))
+                    return this.on_error(error);
+                }
+                const sub = (packet as mqtt.ISubscriptionGrant[])[0];
+                resolve({ topic: sub.topic, qos: sub.qos });
```

Comment:
Looking a little harder since you asked about failing tests...
if we get back sub.qos==128, that means the subscription failed.

For single-topic subscribe(), aws-crt-python turns that into an error, instead of passing it through the QoS.

But for resubscribe_existing_topics() (which is a multi-topic subscribe under the hood), aws-crt-python passes None for the QoS to indicate that a specific topic failed.